core/txpool/txpool.go

```diff
@@ -87,18 +92,63 @@ func New(gasTip *big.Int, chain BlockChain, subpools []SubPool) (*TxPool, error)
 		reservations: make(map[common.Address]SubPool),
 		quit:         make(chan chan error),
 	}
-	for i, subpool := range subpools {
-		if err := subpool.Init(gasTip, head, pool.reserver(i, subpool)); err != nil {
-			for j := i - 1; j >= 0; j-- {
-				subpools[j].Close()
-			}
+	if chain.HasState(head.Root) {
+		if err := pool.init(gasTip, head); err != nil {
 			return nil, err
 		}
+		go pool.loop(head, chain)
+	} else {
+		go pool.lazyInit(gasTip, chain)
```

Comment:
We've discussed this and decided to allow initialising the txpool with "invalid" state instead, by defaulting to the empty state. That would keep the pool "functional" instead of introducing a lazy lifecycle thing. It seems preferable long term as there can be less surprises (i.e. a tx might be rejected / no events might be generated... but no weird crash/hang due to accessing something prior to the lazy init (in the current or future code)).